relay-server/src/endpoints/project_configs.rs

```diff
@@ -172,14 +151,20 @@ async fn get_project_configs(
     Ok(Json(GetProjectStatesResponseWrapper { configs, pending }))
 }
 
-pub fn configure_app(app: App<ServiceState>) -> App<ServiceState> {
-    app.resource("/api/0/relays/projectconfigs/", |r| {
-        r.name("relay-projectconfigs");
-        r.post()
-            .filter(VersionPredicate)
-            .with_async(|b, v| Box::pin(get_project_configs(b, v)).compat());
+fn predicate(Query(query): Query<VersionQuery>) -> bool {
+    query.version >= ENDPOINT_V2 && query.version <= ENDPOINT_V3
+}
 
-        // Forward all unsupported versions to the upstream.
-        r.post().f(crate::endpoints::forward::forward_compat);
+pub async fn handle<B>(state: ServiceState, mut req: Request<B>) -> Result<impl IntoResponse>
+where
+    B: axum::body::HttpBody + Send + 'static,
+    B::Data: Send,
+    B::Error: Into<axum::BoxError>,
+{
+    let data = req.extract_parts().await?;
+    Ok(if predicate(data) {
+        inner.call(req, state).await
```

Comment:
nit: Why is the inner function called `inner` now, and why do we call it with `call`?

```suggestion
        get_project_configs(req, state).await
```